### ISSUE TITLE
Pyright crashes starting from version 1.1.367

### ISSUE THREAD

**Issue by aholmes** [microsoft/pyright#8228]
**Describe the bug**
Starting from version 1.1.367, Pyright seems to crash with an inexplicable error and no clear information regarding the cause. I first noticed it in this GitHub Actions workflow [run](https://github.com/uclahs-cds/BL_Python/actions/runs/9669343046/job/26675663422?pr=78), however, the error occurs on my development machine in a Python venv as well.

```sh
An internal error occurred while type checking file "/home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py": Error: Debug Failure. False expression.
```

<details>
  <summary>Pyright output details</summary>

```sh
$ pyright
WARNING: there is a new pyright version available (v1.1.367 -> v1.1.369).
Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

An internal error occurred while type checking file "/home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py": Error: Debug Failure. False expression.
    at computeMroLinearization (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:3178:21)
    at createClassFromMetaclass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:13116:14)
    at validateCallForInstantiableClass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9810:37)
    at validateCallArgumentsForSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9428:28)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9312:44)
    at useSpeculativeMode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20735:20)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3912:43)
    at doForEachSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:770:9)
    at expandSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3896:13)
    at mapSubtypesExpandTypeVars (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3954:13)
    at validateCallArguments (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9305:26)
    at getTypeOfCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8102:36)
    at getTypeOfExpressionCore (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1157:30)
    at getTypeOfExpression (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1060:26)
    at getTypeOfArgument (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20475:28)
    at evaluateCastCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:10101:28)
    at validateCallForOverloadedFunction (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9683:34)
    at validateCallArgumentsForSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9408:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9312:44)
    at useSpeculativeMode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20735:20)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3912:43)
    at doForEachSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:770:9)
    at expandSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3896:13)
    at mapSubtypesExpandTypeVars (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3954:13)
    at validateCallArguments (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9305:26)
    at getTypeOfCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8102:36)
    at getTypeOfExpressionCore (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1157:30)
    at getTypeOfExpression (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1060:26)
    at evaluateTypesForAssignmentStatement (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16208:39)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19773:25)
    at a.evaluateTypesForStatement [as timeOperation] (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.visitAssignment (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:1182:25)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:367:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at forEach (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at q.walkMultiple (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q._walkStatementsAndReportUnreachable (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2834:18)
    at q.visitSuite (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:288:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q.visitFunction (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:650:18)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:451:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q._walkStatementsAndReportUnreachable (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2834:18)
    at q.visitSuite (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:288:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q.visitClass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:313:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:388:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
Error performing analysis: Error: Debug Failure. False expression.
    at computeMroLinearization (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:3178:21)
    at createClassFromMetaclass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:13116:14)
    at validateCallForInstantiableClass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9810:37)
    at validateCallArgumentsForSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9428:28)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9312:44)
    at useSpeculativeMode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20735:20)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3912:43)
    at doForEachSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:770:9)
    at expandSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3896:13)
    at mapSubtypesExpandTypeVars (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3954:13)
    at validateCallArguments (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9305:26)
    at getTypeOfCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8102:36)
    at getTypeOfExpressionCore (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1157:30)
    at getTypeOfExpression (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1060:26)
    at getTypeOfArgument (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20475:28)
    at evaluateCastCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:10101:28)
    at validateCallForOverloadedFunction (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9683:34)
    at validateCallArgumentsForSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9408:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9312:44)
    at useSpeculativeMode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:20735:20)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9309:24)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3912:43)
    at doForEachSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:770:9)
    at expandSubtype (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3896:13)
    at mapSubtypesExpandTypeVars (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:3954:13)
    at validateCallArguments (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:9305:26)
    at getTypeOfCall (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:8102:36)
    at getTypeOfExpressionCore (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1157:30)
    at getTypeOfExpression (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:1060:26)
    at evaluateTypesForAssignmentStatement (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:16208:39)
    at callback (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/typeEvaluator.ts:19773:25)
    at a.evaluateTypesForStatement [as timeOperation] (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/common/timing.ts:40:20)
    at q.visitAssignment (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:1182:25)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:367:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at forEach (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:924:22)
    at Array.forEach (<anonymous>)
    at q.walkMultiple (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:922:15)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:917:18)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q._walkStatementsAndReportUnreachable (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2834:18)
    at q.visitSuite (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:288:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q.visitFunction (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:650:18)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:451:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q._walkStatementsAndReportUnreachable (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:2834:18)
    at q.visitSuite (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:288:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:544:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:279:19)
    at q.visitClass (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/checker.ts:313:14)
    at q.visit (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:388:29)
    at q.visitNode (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:933:21)
    at q.walk (/home/aholmes/.cache/pyright-python/1.1.367/node_modules/pyright/dist/pyright-internal/src/analyzer/parseTreeWalker.ts:915:37)
```
</details>

**Code or Screenshots**
The crash seems to be triggered by [this file](https://github.com/uclahs-cds/BL_Python/blob/main/src/programming/BL_Python/programming/config/__init__.py).

**VS Code extension or command-line**
This occurs on the command line. VSCode with Pylance seems to be fine.

**Comment by erictraut** [comment]
This is a duplicate of https://github.com/microsoft/pyright/issues/8191, and it was fixed in 1.1.369.

**Comment by aholmes** [comment]
@erictraut it seems that this is still an issue?

```sh
$ npx pyright --version && npx pyright /home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py
pyright 1.1.369
An internal error occurred while type checking file "/home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py": Error: Debug Failure. False expression.
    at computeMroLinearization (/home/aholmes/repos/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:3162:21)
    ...
```

```sh
$ . .venv/bin/activate
(.venv) $ pyright --version && pyright /home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py
pyright 1.1.369
An internal error occurred while type checking file "/home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py": Error: Debug Failure. False expression.
    at computeMroLinearization (/home/aholmes/.cache/pyright-python/1.1.369/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:3162:21)
    ...
```

```sh
(.venv) $ PYRIGHT_PYTHON_FORCE_VERSION=latest pyright /home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py
An internal error occurred while type checking file "/home/aholmes/repos/BL_Python/src/programming/BL_Python/programming/config/__init__.py": Error: Debug Failure. False expression.
    at computeMroLinearization (/home/aholmes/.cache/pyright-python/1.1.369/node_modules/pyright/dist/pyright-internal/src/analyzer/typeUtils.ts:3162:21)
    ...
```

The error persists after running both `python -m pip cache purge` and `npm cache clean --force` and re-making my venv.

**Comment by erictraut** [comment]
Reopening to investigate further. I wasn't able to repro the original bug report, but I thought I understood the problem enough to address it. Evidently my analysis was incomplete. I'll take another swing at it. Now that I have your repro, it should be easier to confirm a fix.

**Comment by aholmes** [comment]
Thanks for checking this out again! Please let me know if I can provide additional information.

**Comment by aholmes** [comment]
I took a quick look at the fix. Am I understanding correctly that the assertion failed because the computed type:
1. is "not a class"
2. has a known type (meaning not Any or Unknown)

If this is correct, are my type annotations in the affected file correct? I'm wondering whether I got something wrong that caused this, or if it was just an unforeseen case in Pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.370
